websites/management/commands/markdown_cleaning/rootrelative_urls.py

```diff
@@ -157,45 +163,66 @@ def fuzzy_find_linked_content(self, url: str):
                 ) from error
             raise self.NotFoundError("Content not found.") from error
 
-    def replace_match(self, match: re.Match, website_content: WebsiteContent) -> str:
+    def replace_match(
+        self, s, l, toks: LinkParseResult, website_content: WebsiteContent
+    ) -> str:
         Notes = self.ReplacementNotes
-        original_text = match[0]
-        url = match.group("url")
-        image_prefix = match.group("image_prefix")
-        is_image = image_prefix == "!"
-        title = match.group("title")
+        original_text = toks.original_text
+        link = toks.link
+        url = urlparse(link.destination)
+        is_image = link.is_image
+        text = link.text
+
+        if not re.match(R".?/?(course|resource)", url.path):
+            return original_text, Notes("Not rootrelative link", None, None)
 
         try:
-            linked_content, note = self.fuzzy_find_linked_content(url)
+            linked_content, note = self.fuzzy_find_linked_content(url.path)
         except self.NotFoundError as error:
             note = str(error)
-            return original_text, Notes(note, is_image, same_site=False)
+            return original_text, Notes(note, is_image, same_site=None)
 
         same_site = linked_content.website_id == website_content.website_id
-        fragment = urlparse(url).fragment
+        fragment = url.fragment
 
         notes = Notes(note, is_image, same_site)
         if same_site:
-            uuid = linked_content.text_id
-
-            if is_image:
-                replacement = f"{{{{< resource {uuid} >}}}}"
-            elif fragment:
-                replacement = (
-                    f'{{{{% resource_link {uuid} "{title}" "#{fragment}" %}}}}'
+            try:
+                if is_image:
+                    shortcode = ShortcodeTag.resource(uuid=linked_content.text_id)
+                else:
+                    shortcode = ShortcodeTag.resource_link(
+                        uuid=linked_content.text_id, text=text, fragment=fragment
+                    )
+                return shortcode.to_hugo(), notes
+            except ValueError:
+                # This happens with within-site links to the homepage.
+                # The text_id of the resource is "sitemetadata", which is not
+                # a uuid. The resource link shortcode probably works in this
+                # case, but there are only 3 of these, so let's not worry about
+                # it. Plus, keeping resource_links as true UUIDs will be nice
+                # if we implement cross-site resource_links down the road.
```

Comment:
@gumaerc Curious if you know whether there's a reason that course homepages have `"sitemetadata"` as their text_id rather than some uuid like everything else.

Reply:
So this is because the file behind this, `course.json`, is generated during a run of `ocw-to-hugo`.  There's no equivalent file in Plone, hence no UUID.  If we generated a new UUID, it would be different during every run of `ocw-to-hugo`.  When we were regularly running `import_ocw_course_sites`, we needed a way to make sure that changes to the metadata file were updated instead of creating a new `WebsiteContent` object every time.